# HG changeset patch
# Parent 85058febc830a8d4f97b0a746dee05a4e501d252

Dont fail empty barrier requests.

Forward ported from
 http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/876a5aaac026

diff -r 85058febc830 drivers/xen/blkback/blkback.c
--- a/drivers/xen/blkback/blkback.c
+++ b/drivers/xen/blkback/blkback.c
@@ -237,8 +237,14 @@ blkif_translate(blkif_t *blkif, struct p
 		return -EACCES;
 
 	size = blkif_size(blkif);
-	if (unlikely(req->sector_number + req->nr_sects > size))
-		return -EACCES;
+	if (likely(req->nr_sects)){
+		blkif_sector_t end = req->sector_number + req->nr_sects;
+
+		if (unlikely(end < req->sector_number))
+			return -EACCES;
+		if (unlikely(end > size))
+			return -EACCES;
+	}
 
 	req->dev  = bdev->bd_inode->i_rdev;
 	req->bdev = bdev;
